staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -95,6 +95,30 @@ type Reflector struct {
 	// etcd, which is significantly less efficient and may lead to serious performance and
 	// scalability problems.
 	WatchListPageSize int64
+	// Called whenever the ListAndWatch drops a connection, whether or not
+	// the connection had an error. So that clients can handle errors
+	// and implement backoff.
+	dropWatchHandler DropWatchHandler
+}
+
+type DropWatchHandler func(err error, doneCh <-chan struct{})
+
+func createDefaultDropWatchHandler(r *Reflector) DropWatchHandler {
```

Comment:
I'd change the `DropWatchHandler` type to `func(r *Reflector, err error, doneCh <- chan struct{})` and then define `func DefaultDropWatchHandler(...)`. Seems more idiomatic to me and having access to Reflector in custom handlers seems useful anyway.

Reply:
done!